### PR TITLE
[HttpFoundation] Add method `getString` to `ParameterBag`

### PR DESCRIPTION
**Pull request by SHTIKOV**
| Q             | A
| ------------- | ---
| Branch?       | 6.1
| Bug fix?      | no
| New feature?  | yes 
| Deprecations? | no 
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

Added method `getString` to `ParameterBag` for use strict type. We really need this method to make it easier to work with strong typing, because the `get` method returns several data types, and we need a specific one.

Reopen old PR https://github.com/symfony/symfony/pull/44787, I closed by my mistake...

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 CHANGELOG
 =========
 
+6.2
+---
+
+ * Add method `convertString` to `ParameterBag`
+ * Rename method `getInt` to `convertInt` in `ParameterBag`
+
```
Comment (FabienPapet) [inline src/Symfony/Component/HttpFoundation/CHANGELOG.md:9]: maybe add deprecate convertInt ?

Comment (SHTIKOV) [reply src/Symfony/Component/HttpFoundation/CHANGELOG.md:9]: but `convertInt` is not deprecated...?

Comment (FabienPapet) [reply src/Symfony/Component/HttpFoundation/CHANGELOG.md:9]: I meant getInt()

**Comment by SHTIKOV** [comment]
@fabpot , WDT?)

**Comment by nicolas-grekas** [comment]
Closing in favor of #44787, which I'm reopening because there are useful comments there that shouldn't be lost.

**Pull request closed without merging**